pkg/manager-server/file_system_lvm.go

```diff
@@ -214,7 +234,61 @@ func (f *FileSystemLvm) Delete() error {
 }
 
 func (f *FileSystemLvm) Mount(mountpoint string) error {
-	return f.mount(f.devPath(), mountpoint, "", []string{"--bind"})
+	// Make the parent directory
+	if err := os.MkdirAll(filepath.Dir(mountpoint), 0755); err != nil {
+		return err
+	}
+
+	// Create an empty file to bind mount the device onto
+	if err := os.WriteFile(mountpoint, []byte(""), 0644); err != nil {
+		return err
+	}
+
+	mounter := mount.New("")
+	mounted, err := mounter.IsMountPoint(mountpoint)
+	if err != nil {
+		return err
+	}
+
+	if !mounted {
+		if err := mounter.Mount(f.devPath(), mountpoint, "none", []string{"bind"}); err != nil {
+			log.Errorf("Mount failed: %v", err)
+			return err
+		}
+
+		if err := os.Chown(f.devPath(), f.uid, f.gid); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (f *FileSystemLvm) Unmount(mountpoint string) error {
+	if mountpoint == "" {
+		return nil
+	}
+
+	mounter := mount.New("")
+	mounted, err := mounter.IsMountPoint(mountpoint)
```

Comment:
The [lustre-csi-driver handles this in a slightly better way ](https://github.com/HewlettPackard/lustre-csi-driver/blob/master/pkg/lustre-driver/service/node.go#L106-L118)- it instead uses `mount.IsNotMountPoint()` which lets you get ride of the special `fs.ErrNotExist` handling, since `IsNotMountPoint()` would just return (false, nil) in that case. Negates the logic in line 282 however.